### PR TITLE
Makes ILabShell optional in toc extension

### PR DESCRIPTION
**Pull request by jweill-aws**
## References

#11419 

## Code changes

Per suggestion by @jtpio in https://github.com/jupyterlab/retrolab/issues/272#issuecomment-963475247, make the `toc` extension optionally dependent on `ILabShell` and, where `ILabShell` is not required, use `app.shell` to refer to the shell (an instance of `IShell`).

## User-facing changes

None expected.

## Backwards-incompatible changes

None expected.

### CODE REVIEW

File: packages/toc-extension/src/index.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ async function activateTOC(
   app: JupyterFrontEnd,
   docmanager: IDocumentManager,
   editorTracker: IEditorTracker,
-  labShell: ILabShell,
+  labShell: ILabShell | null,
```
Comment (jtpio) [inline packages/toc-extension/src/index.ts:69]: The order of the dependencies is relevant, so we need to move that line down below `translator`.

**Review by fcollonval — APPROVED** [review]
LGTM thanks @jweill-aws

**Comment by jtpio** [comment]
Thanks!

**Comment by jtpio** [comment]
This can also be backported to 3.2.x as it should be backward compatible.

**Comment by jtpio** [comment]
@meeseeksdev please backport to 3.2.x

**Pull request merged**